blockstore/splitstore/splitstore.go

```diff
@@ -260,8 +284,14 @@ func (s *SplitStore) Has(ctx context.Context, cid cid.Cid) (bool, error) {
 		if has {
 			return s.has(cid)
 		}
-
-		return s.cold.Has(ctx, cid)
+		switch s.compactType {
+		case hot:
+			return s.cold.Has(ctx, cid)
+		case cold:
+			return s.hot.Has(ctx, cid)
```

Comment:
I don't think so.  Tracking references for protection won't have any effect because we are already in the critical section which begins by taking all protected refs and adding to the markset.

There is no correctness issue because we respect the markset's view of the compacting store.  For example if it is the hot store that is compacting and the markset doesn't have this cid then we only return what the cold store has and so don't leak an incorrect transient has.